network-conversion-server/src/main/java/com/powsybl/iidm/converter/server/IidmConverterService.java

```diff
@@ -0,0 +1,132 @@
+
+    @Autowired
+    private NetworkStoreService networkStoreService;
+
+    @Autowired
+    public IidmConverterService(@Value("${backing-services.case.base-uri:http://case-server/}") String caseServerBaseUri,
+                                @Value("${backing-services.geo-data.base-uri:http://geo-data-server/}") String geoDataBaseUri) {
+        RestTemplateBuilder restTemplateBuilder = new RestTemplateBuilder();
+        caseServerRest = restTemplateBuilder.build();
+        caseServerRest.setUriTemplateHandler(new DefaultUriBuilderFactory(caseServerBaseUri));
+        this.caseServerBaseUri = caseServerBaseUri;
+
+        restTemplateBuilder = new RestTemplateBuilder();
+        geoDataRest = restTemplateBuilder.build();
+        geoDataRest.setUriTemplateHandler(new DefaultUriBuilderFactory(geoDataBaseUri));
+        this.geoDataBaseUri = geoDataBaseUri;
+    }
+
+    NetworkIds persistentStore(String caseName) {
+        byte[] networkByte = getCaseAsByte(caseName);
+        String[] baseName = caseName.split("\\.");
+        ReadOnlyMemDataSource readOnlyMemDataSource = new ReadOnlyMemDataSource(baseName[0]);
+        readOnlyMemDataSource.putData(caseName, networkByte);
+        Network network = networkStoreService.importNetwork(readOnlyMemDataSource);
+        UUID networkUuid = networkStoreService.getNetworkUuid(network);
+        precalculateLines(networkUuid);
+        return new NetworkIds(networkUuid, network.getId());
+    }
+
+    private void precalculateLines(UUID idNetwork) {
+        HttpHeaders requestHeaders = new HttpHeaders();
+        HttpEntity requestEntity = new HttpEntity(requestHeaders);
+
+        Map<String, Object> urlParams = new HashMap<>();
+        urlParams.put("idNetwork", idNetwork);
+
+        UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromHttpUrl(geoDataBaseUri + "/" + IIDM_GEO_DATA_API_VERSION +
+                "/precalculate-lines/{idNetwork}").uriVariables(urlParams);
+
+        geoDataRest.exchange(uriBuilder.toUriString(),
+               HttpMethod.GET,
+               requestEntity,
+               String.class);
+    }
+
+    byte[] getCaseAsByte(String caseName) {
+        HttpHeaders requestHeaders = new HttpHeaders();
+        HttpEntity requestEntity = new HttpEntity(requestHeaders);
+
+        Map<String, Object> urlParams = new HashMap<>();
+        urlParams.put("caseName", caseName);
+
+        UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromHttpUrl(caseServerBaseUri + "/" + CASE_API_VERSION + "/case-server/cases/{caseName}")
+                .uriVariables(urlParams);
+
+        try {
+            ResponseEntity<byte[]> responseEntity = caseServerRest.exchange(uriBuilder.toUriString(),
+                    HttpMethod.GET,
+                    requestEntity,
+                    byte[].class);
+            return responseEntity.getBody();
+        } catch (HttpStatusCodeException e) {
+            LOGGER.error("getCaseAsByte HttpStatusCodeException :", e);
```

Comment:
Either log the exception, or rethrow it, but not both (this leads to duplicate logs)